src/filters/authfilter.bal

```diff
@@ -0,0 +1,111 @@
+import ballerina/config;
+import ballerina/http;
+import ballerina/log;
+import ballerina/stringutils;
+
+
+
+public type AuthFilter object {
+
+    private http:Client clientEndpoint;
+
+    public function __init() {
+        self.clientEndpoint = new (config:getAsString("mosip.auth.base_url"));
+    }
+    *http:RequestFilter;
+
+    public function filterRequest(http:Caller caller, http:Request request,
+        http:FilterContext context) returns boolean {
+        if (config:getAsString("mosip.auth.filter_status", "disable") == "enable") {
+            http:Response errorsResponse = new;
+            if (request.getQueryParamValue("hub.mode") == "publish") {
```

Comment:
Are we checking only for publish, for subscription are we failing the auth?

Reply:
Let me add a condition for Sub also. Discussion of sub was unclear when we discussed with that team but let me add one condition and will add the same logic there.

Reply:
@urviljoshi We should separate permission/role for publishing and subscribing to a topic. Someone authorized to only subscribe should be allowed to publish